File: _정답코드/4주차(임시)/03_하노이_타워_2/solution.cpp
#include <vector>
using namespace std;

vector<vector<int>> answer;

void hanoi(int n, int start, int end, int mid) {
    vector<int> move = {n, start, end};
    if (n == 1) {
        answer.push_back(move);
    } else {
        hanoi(n - 1, start, mid, end);
        answer.push_back(move);
        hanoi(n - 1, mid, end, start);
    }
}

vector<vector<int>> solution(int n) {
    answer.clear();
    hanoi(n, 1, 3, 2);
    return answer;
}

File: _정답코드/4주차(임시)/03_하노이_타워_2/Solution.java
import java.util.*;

public class Solution {
    private static List<int[]> answer = new ArrayList<>();

    public int[][] solution(int n) {
        hanoi(n, 1, 3, 2);
        return answer.toArray(new int[answer.size()][3]);
    }

    private void hanoi(int n, int start, int end, int mid){
        int[] move = {n, start, end};
        if(n == 1){
            answer.add(move);
        } else {
            hanoi(n-1, start, mid, end);
            answer.add(move);
            hanoi(n-1, mid, end, start);
        }
    }
}

